### ISSUE TITLE
Document incorrect about 'default_box_attr' behaviour

### ISSUE THREAD

**Issue by zhaowb** [cdgriffith/Box#130]
In document 
> default_box_attr will first check if it is callable, and will call the object if it is, otherwise it will see if has the copy attribute and will call that, lastly, will just use the provided item as is

But due to Box [__init__](https://github.com/cdgriffith/Box/blob/master/box/box.py#L200)
```
        self._box_config.update({
            'default_box': default_box,
            'default_box_attr': default_box_attr or self.__class__,
```
When  default_box_attr is logical false ('', [], {} etc) it's actually replaced by Box.

This code won't work as expected:
```
x = Box(default_box=True, default_box_attr='')
x.abc  # returns a Box
```
It has to be
```
x = Box(default_box=True, default_box_attr=str)
x.abc  # returns ''
```
So `list` for `[]`, `int` for `0` etc

At least version 3.4.6 is working with `default_box_attr=''`.

Suggestion:
Either change document to indicate this detail, or change code to accept `'', [], 0` etc.
Possible fix is
```
            'default_box_attr': self.__class__ if default_box_attr is None else default_box_attr,
```

**Comment by cdgriffith** [comment]
Thank you both for the reports and the ideas! I already have a `NO_DEFAULT` object I just need to use in this case, that I totally overlooked for some reason. Updated code will look like: 

```
self.__class__ if default_box_attr is NO_DEFAULT else default_box_attr
```

**Comment by cdgriffith** [comment]
With developing this, this is actually going to be a larger change than first expected, as it will change the nature of `default_box`. 

Currently, when an item is retrieved that doesn't exist, it returns the item, but does not set the box to have that item. But then you can run into some bad experiences: 

```
        bx5 = Box(default_box=True, default_box_attr=[])
        assert isinstance(bx5.empty_list_please, list)
        assert len(bx5.empty_list_please) == 0
        bx5.empty_list_please.append(1)
        assert bx5.empty_list_please[0] == 1
```

Also, it is incongruous with the built in `defaultdict` which will set any item that is retrieved: 

```
a = defaultdict(lambda: 5)
a
# defaultdict(<function <lambda> at 0x041285C8>, {})
a['test']
# 5
a
# defaultdict(<function <lambda> at 0x041285C8>, {'test': 5})
```

So this will need to change to save on retrieval. This won't slow down a release, but it does bump it up to a full dot release instead of bugfix.

**Comment by cdgriffith** [comment]
First attempt at fix of this in https://github.com/cdgriffith/Box/commit/1eb54789393063093de4ad1847cabbb51efc2d3b

Pinging @iordanivanov on this thread as well now that there is a possible working update. 

More test cases are needed for this to confirm I'm not missing anything.

**Comment by cdgriffith** [comment]
slated for 4.1 https://github.com/cdgriffith/Box/pull/132

**Comment by cdgriffith** [comment]
Addressed in [4.1.0](https://github.com/cdgriffith/Box/tree/4.1.0)
